File: hw08-spring-mongo-db/src/main/java/ru/otus/hw/models/Comment.java
package ru.otus.hw.models;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

@Getter
@Setter
@RequiredArgsConstructor
@AllArgsConstructor
@Document(collection = "comments")
public class Comment {

    @Id
    private String id;

    private String text;

    @DBRef
    private Book book;

}

File: hw18-spring-cloud/frontend/src/app/model/comment.ts
import {Book} from "./book";

export class Comment {
  id?: number;
  text?: string;
  book?: Book;

  constructor(id?: number,
              text?: string,
              book?: Book) {
    this.id = id;
    this.text = text;
    this.book = book;
  }
}
